main.tf

```diff
@@ -1,13 +1,76 @@
-terraform {
-  required_version = ">= 1.3.0"
-  required_providers {
-    aws = {
-      source  = "hashicorp/aws"
-      version = ">= 4.0"
+locals {
+  runner_run_as = "runners"
+
+  runner_base_config = { # Base configuration shared between arm and amd runners
+    fifo                = true
+    delay_webhook_event = 0
+
+    runner_config = {
+      create_service_linked_role_spot     = true
+      instance_allocation_strategy        = var.instance_allocation_strategy
+      runners_maximum_count               = var.runners_maximum_count
+      enable_organization_runners         = true
+      runner_group_name                   = var.runner_group_name
+      runner_iam_role_managed_policy_arns = var.runner_iam_role_managed_policy_arns
+      runner_os                           = "linux"
+      runner_extra_labels                 = "ondemand"
+      enable_ssm_on_runners               = true
+      idle_config                         = var.idle_config
+      runner_run_as                       = local.runner_run_as
+      userdata_template                   = "${path.module}/templates/user_data.sh"
+      ami_owners                          = ["137112412989"] # amazon
+
+      block_device_mappings = [
+        {
+          device_name           = "/dev/xvda"
+          delete_on_termination = true
+          volume_type           = "gp3"
+          volume_size           = var.volume_size
+          iops                  = null
+          encrypted             = false
+          kms_key_id            = null
+          snapshot_id           = null
+          throughput            = null
+        }
+      ]
+
+      runner_log_files = var.runner_log_files
     }
   }
-}
 
-provider "aws" {
-  region = "eu-central-1"
+  linux_x64 = merge(local.runner_base_config, { # merge of architecture dependant configuration
+    runner_config = merge(local.runner_base_config.runner_config, {
+      runner_architecture = "x64"
+      instance_types      = var.amd_instance_types
+    })
+    matcherConfig = {
+      labelMatchers = [["self-hosted", "linux", "x64", "ondemand"]]
```

Comment:
i think this needs to be further parametrized (add a team label?)  if you would like to use this for different teams inside a shared GitHub organization - otherwise all the runners from the different team AWS will show up with the same labels and when execution the action you can end up on a "wrong" one .. for unity feed is was OK so far because they use the same AWS infra / environments/ permissions ....

Reply:
![image](https://user-images.githubusercontent.com/93534965/223170839-f57bb659-19bb-4791-b7aa-296eb9d2623e.png)